### PR TITLE
chore: log module informations

### PR DESCRIPTION
**Pull request by dmehala**
# Description
Log informations about the module state, like, the version of the module and its dependencies, features available etc.

Screenshot 2024-07-12 at 23 45 27

### CODE REVIEW

File: src/ngx_http_datadog_module.cpp [hunk 1]
```diff
@@ -423,6 +425,49 @@ ngx_module_t ngx_http_datadog_module = {
...
+    case NGX_HTTP_POST_ACCESS_PHASE:
+      return "NGX_HTTP_POST_ACCESS_PHASE";
+    case NGX_HTTP_PRECONTENT_PHASE:
+      return "NGX_HTTP_PRECONTENT_PHASE";
+    case NGX_HTTP_CONTENT_PHASE:
+      return "NGX_HTTP_CONTENT_PHASE";
+    case NGX_HTTP_LOG_PHASE:
+      return "NGX_HTTP_LOG_PHASE";
+  }
+  return "UNKNOWNED_PHASE";
```
Comment (dmehala) [inline src/ngx_http_datadog_module.cpp:453]:
```suggestion
  return "UNKNOWN_PHASE";
```

File: src/ngx_http_datadog_module.cpp [hunk 2]
```diff
@@ -423,6 +425,49 @@ ngx_module_t ngx_http_datadog_module = {
...
+}
+
+template <typename F>
```
Comment (dmehala) [inline src/ngx_http_datadog_module.cpp:456]: - [ ] Remove template

File: src/ngx_http_datadog_module.cpp [hunk 3]
```diff
@@ -423,6 +425,49 @@ ngx_module_t ngx_http_datadog_module = {
...
+static int set_handler(ngx_log_t *log,
+                       ngx_http_core_main_conf_t *core_main_conf,
+                       ngx_http_phases phase, F callback) {
+  auto handler = static_cast<ngx_http_handler_pt *>(
+      ngx_array_push(&core_main_conf->phases[phase].handlers));
+  if (handler == nullptr) return NGX_ERROR;
+  *handler = callback;
+
+  ngx_log_error(NGX_LOG_NOTICE, log, 0,
+                "nginx-datadog: installed handler on {} phase",
```
Comment (dmehala) [inline src/ngx_http_datadog_module.cpp:466]:
```suggestion
                "nginx-datadog: installed handler on %s phase",
```

File: nginx_module.ld [hunk 4]
```diff
@@ -4,5 +4,6 @@
 		ngx_module_names;
 		ngx_module_order;
 		datadog_version_*;
+    datadog_semver_*;
 	local: *; /* Hide all other symbols */
```
Comment (pablomartinezbernardo) [inline nginx_module.ld:8]:
```suggestion
		ngx_module_names;
		ngx_module_order;
		datadog_version_*;
		datadog_semver_*;
	local: *; /* Hide all other symbols */
```

File: nginx_module.ld-prime [hunk 5]
```diff
@@ -3,6 +3,13 @@
 _ngx_modules
```
Comment (pablomartinezbernardo) [inline nginx_module.ld-prime:3]: I see `datadog_build_id_nginx_mod` is not being exported, is there a reason for that?

**Review by pablomartinezbernardo — APPROVED** [review]

File: nginx_module.ld-prime (hunk 5 above)
Comment (dmehala) [reply nginx_module.ld-prime:3]: addressed in 68a4f0da82bd62d73f2e80627377d4786e102a91

**Pull request merged**